pkg/client/cli/cmds_misc.go

```diff
@@ -111,11 +112,16 @@ func InitConnectRequest(cmd *cobra.Command) (*connector.ConnectRequest, *pflag.F
 		`Overrides any other manager namespace set in config`)
 	flags.AddFlagSet(nwFlags)
 
-	kubeConfig := genericclioptions.NewConfigFlags(false)
-	kubeConfig.Namespace = nil // "connect", don't take --namespace
-	kubeFlags := pflag.NewFlagSet("Kubernetes flags", 0)
-	kubeConfig.AddFlags(kubeFlags)
-	flags.AddFlagSet(kubeFlags)
+	// Only include the kubernetes flags if the user daemon is started from the CLI. It's assumed that it is configured
+	// already using a default .kube/config if it is reachable using an address.
+	var kubeFlags *pflag.FlagSet
+	if client.GetEnv(ctx).UserDaemonAddress == "" {
```

Comment:
I'm a little confused. Are we assuming that the daemon is already connected to the cluster if `UserDaemonAddress` is set?

Reply:
No, but we assume that the daemon is started and that the CLI can connect to it.

Reply:
I dont get why we can get rid of the kubeflags if we are not connected yet, Im sure im just missing something 🤷